apps/src/lib/levelbuilder/rubrics/RubricsContainer.jsx

```diff
@@ -1,49 +1,166 @@
+    while (
+      learningGoalList.some(
+        learningGoal =>
+          learningGoal.key === `learningGoal-${learningGoalNumber}`
+      )
+    ) {
+      learningGoalNumber++;
+    }
+
+    return `learningGoal-${learningGoalNumber}`;
+  };
+
+  const addNewConceptHandler = event => {
+    event.preventDefault();
+
+    const newKey = generateLearningGoalKey();
+    const newId = newKey;
+    const startingData = {
+      key: newKey,
+      id: newId,
+      learningGoal: '',
+      aiEnabled: false,
+    };
 
-  const renderLearningGoalItems = learningGoalList.map(goal => (
-    <LearningGoalItem
-      deleteItem={() => deleteKeyConcept(goal.id)}
-      key={goal.id}
-    />
-  ));
+    const oldLearningGoalList = learningGoalList;
+
+    setLearningGoalList([...oldLearningGoalList, startingData]);
+  };
 
   const deleteKeyConcept = id => {
+    event.preventDefault();
     var updatedLearningGoalList = learningGoalList.filter(
       item => item.id !== id
     );
     setLearningGoalList(updatedLearningGoalList);
   };
 
-  const addNewConceptHandler = event => {
-    // temporary tool for creating unique Ids
-    var updatedId = currentId + 1;
+  const handleAiEnabledChange = idToUpdate => {
+    const newLearningGoalData = learningGoalList.map(learningGoal => {
+      if (idToUpdate === learningGoal.id) {
+        const updatedAiValue = !learningGoal.aiEnabled;
+        return {
+          ...learningGoal,
+          ['aiEnabled']: updatedAiValue,
+        };
+      } else {
+        return learningGoal;
+      }
+    });
+    setLearningGoalList(newLearningGoalData);
+  };
 
-    setCurrentId(updatedId);
+  const handleLearningGoalNameChange = (updatedName, idToUpdate) => {
+    const newLearningGoalData = learningGoalList.map(learningGoal => {
+      if (idToUpdate === learningGoal.id) {
+        return {
+          ...learningGoal,
+          ['learningGoal']: updatedName,
+        };
+      } else {
+        return learningGoal;
+      }
+    });
+    setLearningGoalList(newLearningGoalData);
+  };
 
-    // update the goal list
-    setLearningGoalList(oldList => {
-      return [...oldList, {id: updatedId}];
+  // TODO: Check that there is at least one programming level here
+  const initialLevelForAssessment = !!rubric ? rubric.levelId : levels[0].id;
+  const [selectedLevelForAssessment, setSelectedLevelForAssessment] = useState(
+    initialLevelForAssessment
+  );
+
+  const saveRubric = event => {
+    event.preventDefault();
+    const dataUrl = !!rubric ? `/rubrics/${rubric.id}/edit` : RUBRIC_PATH;
+    const method = !!rubric ? 'PATCH' : 'POST';
+
+    // Checking that the csrf-token exists since it is disabled on test
+    const csrfToken = document.querySelector('meta[name="csrf-token"]')
+      ? document.querySelector('meta[name="csrf-token"]').attributes['content']
+          .value
+      : null;
+
+    const learningGoalListAsData = transformKeys(learningGoalList);
```

Comment:
Not sure where to put this comment and this may not be the right place to resolve but one thing that's missing in saving the learning goals is the `position` attribute (ie the order that the learning goals should appear). I think you could either set that attribute here, before sending the learning goals off to the server or on the server side -- up to you.

Reply:
I did add the `position` to the[ key concepts. ](https://github.com/code-dot-org/code-dot-org/pull/53432/files#r1312201215).  It is now saving in the table.

Reply:
Just checking -- we can't rearrange key concepts yet right?